src/common/FrameManager.ts

```diff
@@ -316,24 +358,27 @@ export class FrameManager extends EventEmitter {
     this.emit(FrameManagerEmittedEvents.FrameNavigated, frame);
   }
 
-  async _ensureIsolatedWorld(name: string): Promise<void> {
-    if (this._isolatedWorlds.has(name)) return;
-    this._isolatedWorlds.add(name);
-    await this._client.send('Page.addScriptToEvaluateOnNewDocument', {
+  async _ensureIsolatedWorld(session: CDPSession, name: string): Promise<void> {
+    const key = `${session.id()}:${name}`;
+    if (this._isolatedWorlds.has(key)) return;
+    this._isolatedWorlds.add(key);
```

Comment:
should not we track isolatedWorlds by session then?

Reply:
ah sorry, my bad

Reply:
`key` now includes the session id, so I think this should be addressed already? Or am I missing something?

Reply:
yeah, I have already resolved that conversation.